app/components/search_bar/search_bar.ios.js

```diff
@@ -1,96 +1,145 @@
+    };
 
-        onBlur(event);
+    onChangeText = (text) => {
+        return new Promise((resolve) => {
+            if (this.props.onChangeText) {
+                this.props.onChangeText(text);
+            }
+            resolve();
+        });
     };
 
-    blur = () => {
-        this.searchBar.blur();
+    onDelete = () => {
+        return new Promise((resolve) => {
+            if (this.props.onChangeText) {
+                this.props.onChangeText('');
+            }
+            resolve();
+        });
     };
 
-    searchBarRef = (ref) => {
-        this.searchBar = ref;
+    onFocus = () => {
+        return new Promise((resolve) => {
+            if (this.props.onFocus) {
+                this.props.onFocus();
+            }
+            resolve();
+        });
+    };
+
+    onSearch = (text) => {
+        return new Promise((resolve) => {
+            if (this.props.onSearchButtonPress) {
+                this.props.onSearchButtonPress(text);
+            }
+            resolve();
+        });
     };
 
     render() {
+        if (this.state.placeholderWidth) {
+            return (
+                <Search
+                    {...this.props}
+                    ref='search'
+                    placeholderCollapsedMargin={this.state.placeholderWidth}
+                    placeholderExpandedMargin={20}
+                    searchIconCollapsedMargin={this.state.placeholderWidth + 10}
+                    searchIconExpandedMargin={10}
+                    shadowVisible={false}
+                    onCancel={this.onCancel}
+                    onChangeText={this.onChangeText}
+                    onFocus={this.onFocus}
+                    onSearch={this.onSearch}
+                    afterDelete={this.afterDelete}
+                    onDelete={this.onDelete}
+                />
+            );
+        }
+
         return (
-            <SearchBar
-                ref={this.searchBarRef}
-                barStyle={this.props.barStyle}
-                searchBarStyle={this.props.searchBarStyle}
-                text={this.props.text}
-                placeholder={this.props.placeholder}
-                showsCancelButton={this.state.displayCancelButton}
-                hideBackground={false}
-                textFieldBackgroundColor={this.props.textFieldBackgroundColor}
-                placeholderTextColor={this.props.placeholderTextColor}
-                textColor={this.props.textColor}
-                barTintColor={this.props.barTintColor}
-                tintColor={this.props.tintColor}
-                onChange={this.props.onChange}
-                onChangeText={this.props.onChangeText}
-                onFocus={this.onFocus}
-                onBlur={this.onBlur}
-                onSearchButtonPress={this.props.onSearchButtonPress}
-                onCancelButtonPress={this.props.onCancelButtonPress}
-            />
+            <Text
+                style={{position: 'absolute', top: -3000, left: -1000}}
```

Comment:
Hmm, I wonder if I could do something like this in posts to be able to have a bit more control over text sizing. I feel like I did try something like that before though, so possibly not